knowledge-base/date-input-select-am-pm-on-keypress.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Select AM/PM by pressing A or P on keyboard
+description: How to select AM/PM by pressing A or P on keyboard in TimePicker, DateTimePicker and DateInput
+type: how-to
+page_title: Select AM/PM by pressing A or P on keyboard
+slug: date-input-kb-select-am-pm-on-keypress
+position:
+tags:
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>TimePicker for Blazor, DateTimePicker for Blazor, DateInput for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+When typing in time in the TimePicker/DateTimePicker (enter hours,enter mins, arrow up/down for AM/PM) it would be nice if I could select AM or PM by pressing A or P on the keyboard also.
+
+When using the DateInput component and using format "hh:mm:ss tt" the UI part allows me to edit the hour minute seconds correctly, but the AM / PM is not editable by typing A or P.  Only changes the AM/PM by using up arrow and down arrow on keyboard.  
+
+Any way to make the AM/PM change the way the hh mm ss does by directly entering A or P?
+
+
+## Solution
+
+While A and P might be very common symbols, there are many other symbols that might be used in different countries and cultures. Thus, the component cannot track all such possible combinations, and it exposes the Up and Down Arrows as options to change the time segments, including the AM/PM indicator.
+
+In order to change AM/PM by pressing A or P on keyboard you can get the input events such as keydown. They are not built-in in the component, because exposing so many events like that will be a major performance hit for a small percentage of people who need them. However, you can capture them as they bubble up the DOM in a parent element, as described in this article - [Capture input keyboard events]({%slug inputs-kb-handle-keyboard-events%}).
```

Comment:
the actual reason is that there can be many different abbreviations, and the component can't know them all. Moreover, the user might expect one or the other (.e.g, if their langauge uses C and D they may still want A and P). We already handle the keydown, but we can't listen for a particular symbol